components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java

```diff
@@ -307,6 +309,35 @@ public void deployWebsocketAPI(API api,APIGatewayAdminClient client) throws APIM
 
     }
 
+    /**
+     * add new api version at the API Gateway
+     *
+     * @param artifact
+     * @param api
+     */
+    public void createNewWebsocketApiVersion(GenericArtifact artifact, API api) {
+        try {
+            APIGatewayManager gatewayManager = APIGatewayManager.getInstance();
+            APIGatewayAdminClient client;
+            Set<String> environments = APIUtil.extractEnvironmentsForAPI(
+                    artifact.getAttribute(APIConstants.API_OVERVIEW_ENVIRONMENTS));
+            api.setEndpointConfig(artifact.getAttribute(APIConstants.API_OVERVIEW_ENDPOINT_CONFIG));
+            api.setContext(artifact.getAttribute(APIConstants.API_OVERVIEW_CONTEXT));
+            for (String environmentName : environments) {
+                Environment environment = this.environments.get(environmentName);
+                client = new APIGatewayAdminClient(api.getId(), environment);
+                gatewayManager.deployWebsocketAPI(api, client);
+            }
+        } catch (Exception ex) {
```

Comment:
You have to catch that specific Exception thrown from that code block, not 'Exception'.